src/bundle/Core/Command/VirtualFieldDuplicateFixCommand.php

```diff
@@ -0,0 +1,272 @@
+        $stopwatch = new Stopwatch(true);
+        $stopwatch->start('total', 'command');
+
+        $force = $input->getOption('force');
+        if ($force) {
+            $input->setInteractive(false);
+        }
+
+        $batchSize = (int)$input->getOption('batch-size');
+        if ($batchSize === 0) {
+            $style->warning('Batch size is set to 0. Nothing to do.');
+
+            return Command::INVALID;
+        }
+
+        $maxIterations = (int)$input->getOption('max-iterations');
+        if ($maxIterations === 0) {
+            $style->warning('Max iterations is set to 0. Nothing to do.');
+
+            return Command::INVALID;
+        }
+
+        $sleep = (int)$input->getOption('sleep');
+
+        try {
+            $totalCount = $this->getDuplicatedAttributeTotalCount($style, $stopwatch);
+
+            if ($totalCount > 0) {
+                $confirmation = $this->askForConfirmation($style);
+                if (!$confirmation && !$force) {
+                    $style->info('Confirmation rejected. Terminating.');
+
+                    return Command::FAILURE;
+                }
+            } else {
+                $style->success('Database is clean of attribute duplicates. Nothing to do.');
+
+                return Command::SUCCESS;
+            }
+
+            $iteration = 1;
+            $totalDeleted = 0;
+            do {
+                $deleted = 0;
+                $stopwatch->start('iteration', 'sql');
+
+                $attributes = $this->getDuplicatedAttributesBatch($batchSize);
+                foreach ($attributes as $attribute) {
+                    $attributeIds = $this->getDuplicatedAttributeIds($attribute);
+
+                    $deleted += $this->deleteAttributes($attributeIds);
+                    $totalDeleted += $deleted;
+                }
+
+                $style->info(
+                    sprintf(
+                        'Iteration %d: Removed %d duplicates (total removed this execution: %d). [Debug %s]',
+                        $iteration,
+                        $deleted,
+                        $totalDeleted,
+                        $stopwatch->stop('iteration')
+                    )
+                );
+
+                if ($maxIterations !== self::MAX_ITERATIONS_UNLIMITED && ++$iteration > $maxIterations) {
+                    $style->warning('Max iterations count reached. Terminating.');
+
+                    return self::FAILURE;
+                }
+
+                // Wait, if needed, before moving to next iteration
+                usleep($sleep * 1000);
+            } while ($batchSize === count($attributes));
+
+            $style->success(sprintf(
+                'Operation successful. Removed total of %d duplicates. [Debug %s]',
+                $totalDeleted,
+                $stopwatch->stop('total')
+            ));
+        } catch (Exception $exception) {
+            $style->error($exception->getMessage());
```

Comment:
If you intend to break execution anyway on exception, then allow it to "fall through". You don't need to catch it.